From tj@kernel.org  Mon Jun 27 12:52:44 2011
From: Tejun Heo <tj@kernel.org>
Date: Tue, 14 Dec 2010 06:09:59 +0000
Subject: [PATCH 207/271] bnx2: remove cancel_work_sync() from remove_one
Git-commit: cb8f404893bab40431f7eeb2511454031b07e7df
Patch-mainline: v2.6.38-rc1
References: FATE#311458 FATE#311459 FATE#311460 FATE#311461 FATE#311462 bnc#698050

Upstream commit: cb8f404893bab40431f7eeb2511454031b07e7df

Michael pointed out that bnx2_close() already cancels bp->reset_task
and thus it is guaranteed to be idle when bnx2_remove_one() is called.
Remove the unnecessary cancel_work_sync() in remove_one.

Signed-off-by: Tejun Heo <tj@kernel.org>
Acked-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/bnx2.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/drivers/net/bnx2.c b/drivers/net/bnx2.c
index 3d00594..f46bc17 100644
--- a/drivers/net/bnx2.c
+++ b/drivers/net/bnx2.c
@@ -8457,8 +8457,6 @@ bnx2_remove_one(struct pci_dev *pdev)
 	struct net_device *dev = pci_get_drvdata(pdev);
 	struct bnx2 *bp = netdev_priv(dev);
 
-	cancel_work_sync(&bp->reset_task);
-
 	unregister_netdev(dev);
 
 	if (bp->mips_firmware)
-- 
1.6.4.GIT

